streams/src/main/java/org/apache/kafka/streams/kstream/WindowedSerdes.java

```diff
@@ -46,14 +50,14 @@ public SessionWindowedSerde(final Serde<T> inner) {
     /**
      * Construct a {@code TimeWindowedSerde} object for the specified inner class type.
      */
-    static public <T> Serde<Windowed<T>> timeWindowedSerdeFrom(final Class<T> type) {
+    static public <T> TimeWindowedSerde<T> timeWindowedSerdeFrom(final Class<T> type) {
```

Comment:
The return type `Serde<Windowed<T>>` is defined explicitly for callers to avoid casting. I think we can keep it as is.

Reply:
Kept the old return type.